diff --git a/build/cmake/CMakeLists.txt b/build/cmake/CMakeLists.txt
index 347d41c..9444fea 100644
--- a/build/cmake/CMakeLists.txt
+++ b/build/cmake/CMakeLists.txt
@@ -92,7 +92,7 @@ endif ()
 if (ANDROID)
     set(ZSTD_MULTITHREAD_SUPPORT_DEFAULT OFF)
     # Old versions of bionic libc don't have fseeko/ftello
-    if ((NOT ${ANDROID_PLATFORM_LEVEL}) OR ${ANDROID_PLATFORM_LEVEL} VERSION_LESS 24)
+    if ((NOT DEFINED ANDROID_PLATFORM_LEVEL) OR ANDROID_PLATFORM_LEVEL VERSION_LESS "24")
         message(STATUS "Setting compile definitions for old Android API")
         add_compile_definitions(LIBC_NO_FSEEKO)
     endif ()
